src/type/matrix/utils/matAlgo15xAs.js

```diff
@@ -0,0 +1,26 @@
+import { factory } from '../../../utils/factory.js'
+import { deepMap as map } from '../../../utils/array.js'
+
+const name = 'matAlgo15xAs'
+const dependencies = []
+
+export const createMatAlgo15xAs = /* #__PURE__ */ factory(name, dependencies, () => {
+  /**
+   * Iterates over Array items and invokes the callback function f(Aij..z, b).
+   * Callback function invoked MxN times.
+   *
+   * C(i,j,...z) = f(Aij..z, b)
+   *
+   * @param {Array}    a                 The Array instance (A)
+   * @param {Scalar}   b                 The Scalar value
+   * @param {Function} cf                The f(Aij..z,b) operation to invoke
+   * @param {boolean}  inverse           A true value indicates callback should be invoked f(b,Aij..z)
+   *
+   * @return {Array}                    Array (C)
+   *
+   * https://github.com/josdejong/mathjs/pull/346#issuecomment-97659042
+   */
+  return function matAlgo15xAs (a, b, cf, inverse) {
+    return inverse ? map(a, v => cf(b, v)) : map(a, v => cf(v, b))
```

Comment:
For sufficiently large arrays, is it worth checking the array for type-homogeneity of its leaf entries, and if they are type-homogeneous, doing the appropriate typed.find on cf before calling map?  I guess the same question goes for Algorithm 14: if the matrix does not know if it has a uniform type, is there a size at which it becomes advantageous to first check if the matrix is uniform in type, and specializing the callback function if it is?